cuda/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
 add_library(ginkgo_cuda $<TARGET_OBJECTS:ginkgo_cuda_device> "")
 include(${PROJECT_SOURCE_DIR}/cmake/template_instantiation.cmake)
-add_instantiation_files(. matrix/csr_kernels.instantiate.cu CSR_INSTANTIATE)
+add_instantiation_files(${PROJECT_SOURCE_DIR}/common/cuda_hip matrix/csr_kernels.instantiate.cpp CSR_INSTANTIATE)
```

Comment:
wouldn't this lead to conflicts if both cuda and hip are enabled?

Reply:
I think the issue (if it exists) could be prevented by instantiating the files in the `common/cuda_hip/CMakeLists.txt`

Reply:
No, those are being instantiated in different folders. The reason we need to do it here is related to file properties being scoped to directories, so if we do it in cuda_hip, the dependency between the auto-generation of the file and the library being compiled from it is broken.

Reply:
You're right, they are instantiated in the current binary directory, I was just looking at the wrong place.